sale_website_cleanup/models/sales.py

```diff
@@ -0,0 +1,31 @@
+# coding: utf-8
+# © 2016 Vauxoo - http://www.vauxoo.com
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# info Vauxoo (info@vauxoo.com)
+# coded by: karen@vauxoo.com
+# planned by: nhomar@vauxoo.com
+
+from openerp import api, models
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+
+class CancelOrders(models.Model):
+    """Inherit the sale.order model in order to cancel orders that have
+    created more than 24 hours
+    """
+
+    _inherit = 'sale.order'
+
+    @api.multi
+    def cancel_old_orders(self):
+        sale_order_obj = self.env['sale.order']
+        rel_date = datetime.today() - relativedelta(days=1)
+        strf_date = rel_date.strftime('%Y-%m-%d %H:%M:%S')
+        draft_order = sale_order_obj.search([
+            ('state', '=', 'draft'),
+            ('partner_id.id', '=', self.env.ref('base.public_user')),
+            ('date_order', '<=', strf_date)
+        ])
+        for order in draft_order:
+            order.action_cancel()
```

Comment:
@@ -0,0 +1,31 @@
 +# coding: utf-8
 +# © 2016 Vauxoo - http://www.vauxoo.com
 +# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 +# info Vauxoo (info@vauxoo.com)
 +# coded by: karen@vauxoo.com
 +# planned by: nhomar@vauxoo.com
 +
 +from openerp import api, models
 +from datetime import datetime
 +from dateutil.relativedelta import relativedelta
 +
 +
 +class CancelOrders(models.Model):
 +    """Inherit the sale.order model in order to cancel orders that have
 +    created more than 24 hours
 +    """
 +
 +    _inherit = 'sale.order'
 +
 +    @api.multi
 +    def cancel_old_orders(self):
 +        sale_order_obj = self.env['sale.order']
 +        rel_date = datetime.today() - relativedelta(days=1)
 +        strf_date = rel_date.strftime('%Y-%m-%d %H:%M:%S')
 +        draft_order = sale_order_obj.search([
 +            ('state', '=', 'draft'),
 +            ('partner_id.id', '=', self.env.ref('base.public_user')),
@nhomar     Vauxoo member
nhomar added a note 29 minutes ago
self.env.ref devuelve el browseable.

So.... el dominio es:

 ('partner_id', '=', self.env.ref('base.public_user'))`
 @nhomar     Vauxoo member
nhomar added a note 19 minutes ago
As it fails, try:

 ('partner_id', '=', self.env.ref('base.public_user').id)
 Add a line note
 +            ('date_order', '<=', strf_date)
 +        ])
@nhomar     Vauxoo member

Include here another logger _logger.

_logger.info("Cancelling this order id: %s" % order.id)